### ISSUE TITLE
About the `netket::AbstractGraph::EdgeColorsFrom*` functions

### ISSUE THREAD

**Issue by twesterhout** [netket/netket#83]
Could someone, please, clarify for me what are the pre- and postconditions for these functions?
Looking at the code I have a strange feeling that this whole "color" feature is broken and we don't have issues piling up just because nobody uses it. Really hope I'm wrong and someone more knowledgeable about this topic can explain (pinging you @jamesETsmith as you're the author, sorry :)).

**Comment by jamesETsmith** [comment]
@twesterhout for `EdgeColorsFromList()` we are assuming that two-dimensional `colorlist` vector is Nx3, where N is the number of edges. In `colorlist[0]` our three elements are the label for vertices 1 and 2 and the integer color for that edge (in that order). I realized that the ordering is someone arbitrary here, so in PR #65, when I created the python utilities to convert networkx and igraph graphs into `netket::Graphs` this is taken care of automatically so users don't have to remember the order. 

As for the `EdgeColorsFromAdj()` function I am assuming that the adjacency list is built and stored
```cpp
    // If edge colors are specificied read them in, otherwise set them all to
    // 0
    if (FieldExists(pars, "EdgeColors")) {
      std::vector<std::vector<int>> colorlist =
          FieldVal<std::vector<std::vector<int>>>(pars, "EdgeColors", "Graph");
      EdgeColorsFromList(colorlist, eclist_);
    } else {
      InfoMessage() << "No colors specified, edge colors set to 0 "
                    << std::endl;
      EdgeColorsFromAdj(adjlist_, eclist_);
    }
```

Since it looks like you are going to get rid of the storing the adjacency list (or only storing it temporarily) in #85 we may want to do this a different way.

After both functions are complete the graph member varialbe eclist_ should be populated.
